kyaml/runfn/runfn.go

```diff
@@ -507,7 +507,19 @@ func (r *RunFns) ffp(spec runtimeutil.FunctionSpec, api *yaml.RNode, currentUser
 	}
 
 	if r.EnableExec && spec.Exec.Path != "" {
-		ef := &exec.Filter{Path: spec.Exec.Path}
+		currentDir, err := os.Getwd()
+		if err != nil {
+			return nil, err
+		}
+
+		wd, _, err := kioutil.GetFileAnnotations(api)
+		if err != nil || wd == "" || wd == "/" {
+			wd = currentDir
+		}
```

Comment:
put the above in a function like
```
wd, err := determineWorkingDir()
```
Also, the call to `os.Getwd` should be inside the `if` after the `GetFileAnnotations` call (no need to call the `os` if there's an annotation)

Reply:
Is there a valid case where the annotation will not be present? If the Kustomization is apparently in your root directory (very weird), should that be an error? I feel like falling back on current directory could just be confusing.

Reply:
> Is there a valid case where the annotation will not be present? If the Kustomization is apparently in your root directory (very weird), should that be an error? I feel like falling back on current directory could just be confusing.

Makes sense. We can just return the error.

Reply:
Updated the PR

Reply:
Ah I've discovered the unfortunate case of `kustomize fn run` (tested by the e2e tests in cmd/config), which will not annotate the function config (and it probably shouldn't). I'm reverting this change - if the annotation is missing we should fall back to the current directory.

Reply:
makes sense